src/cache/redis.ts

```diff
@@ -108,4 +110,53 @@ export class RedisCache<T = unknown> implements Cache<T> {
     // Record exec command sent to Redis
     recordRedisCommandMetric(CMD_SENT_STATUS.SUCCESS, 'exec')
   }
+
+  async lock(key: string, cacheLockDuration: number, retryCount: number): Promise<void> {
+    const start = Date.now()
+    const log = (msg: string) => `[${(Date.now() - start) / 1000}]: ${msg}`
+    const durationBuffer = cacheLockDuration * 0.7
```

Comment:
Because we want to cover the case where an adapter crashes without extending the lock, we need the total time spent retrying to necessarily be greater than the time the lock would be extended for.
Given the limitations cause by the redlock bug, I think we should handle retries ourselves which should be very simple, and would make it so there shouldn't be any issues when we acquire the first lock